src/pages/movies/[movieId].js

```diff
@@ -0,0 +1,175 @@
+import styles from "@/styles/movie.module.scss";
+import { getMovie, getShowtimesForMovie } from "@/utils/dbFunctions";
+import dbConnect from "@/utils/dbConnect";
+import Link from "next/link";
+
+export async function getServerSideProps(context) {
+  await dbConnect();
+
+  const { movieId } = context.query;
+  const movie = (await getMovie(movieId)) ?? null;
+
+  if (!movie) {
+    return {
+      notFound: true
+    };
+  }
+
+  const showtimes = await getShowtimesForMovie(movieId);
+
+  return {
+    props: {
+      showtimes: JSON.parse(JSON.stringify(showtimes)).sort(
+        (a, b) => new Date(a.date) - new Date(b.date)
+      ),
+      length: movie.length,
+      title: movie.title,
+      description: movie.description,
+      genre: movie.genre.join(", "),
+      director: movie.director.join(", "),
+      actors: movie.actors.join(", "),
+      country: movie.country,
+      year: new Date(movie.release).getFullYear(),
+      ageLimit: movie.age,
+      imageURL: movie.image
+    }
+  };
+}
+
+const Movie = ({
+  title,
+  description,
+  genre,
+  director,
+  actors,
+  country,
+  year,
+  ageLimit,
+  imageURL,
+  showtimes,
+  length
+}) => {
+  const showtimesList = showtimes.map((showtime) => {
+    const date = new Date(showtime.date);
+    return (
+      <li key={Math.random() * 34231213}>
+        {date
+          .toLocaleDateString("sv-SE", {
+            weekday: "short",
+            day: "numeric",
+            month: "short"
+          })
+          .toUpperCase()}
+        <br />{" "}
+        {date.toLocaleTimeString("sv-SE", {
+          hour: "numeric",
+          minute: "numeric"
+        })}{" "}
+        <span>
+          {showtime.salong}
+          <br /> {length} min
+        </span>
+        <Link
+          href={{
+            pathname: "/booking",
+            query: {
+              showtimeId: showtime["_id"],
+              showtimeSalong: showtime.salong,
+              showtimeDate: showtime.date,
```

Comment:
Maybe I am at fault, but I can't seam to get the Link query to pass a Date object. The only way I got it to work was to convert the date toUTCString() and send it as a string and then convert it back to a date at the booking page.

Reply:
No, that is perfectly normal. It is utilizing the query string in the URL, and those are strings by default.